src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets

```diff
@@ -50,8 +50,7 @@
       Text="RuntimeIdentifier is required for native compilation. Try running dotnet publish with the -r option value specified." />
     <Error Condition="'$(GeneratePackageOnBuild)' == 'true'" Text="GeneratePackageOnBuild is not supported for native compilation." />
     <Error Condition="'$(OutputType)' != 'Library' and '$(NativeLib)' != '' and '$(CustomNativeMain)' != 'true'" Text="NativeLib requires OutputType=Library." />
-    <!-- See https://github.com/dotnet/runtime/issues/89346 for details -->
-    <Error Condition="'$(NativeLib)' != '' and '$(EventSourceSupport)' == 'true'" Text="EventSource is not supported when compiling to a native library. Set EventSourceSupport to false." />
+    <Warning Condition="'$(NativeLib)' != '' and '$(EventSourceSupport)' == 'true' and '$(_SuppressNativeLibEventSourceWarning)' != 'true'" Text="EventSource has known issues when compiling to a native library: see https://github.com/dotnet/runtime/issues/91762 for details and workarounds. It is recommended to leave EventSourceSupport MSBuild property at the default value of false. If you wish to leave EventSourceSupport at true and remove this warning instead, set _SuppressNativeLibEventSourceWarning MSBuild property to true." />
```

Comment:
```suggestion
    <Warning Condition="'$(NativeLib)' != '' and '$(EventSourceSupport)' == 'true' and '$(_SuppressNativeLibEventSourceWarning)' != 'true'" Text="Multiple components (including CoreCLR) with EventSource enabled can't be loaded into the same process: see https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/diagnostics for details and workarounds. EventSource should only be enabled in libraries if you are certain it will be the only component with EventSource loaded into the process. To suppress this warning you can set  the _SuppressNativeLibEventSourceWarning MSBuild property to true." />
```